streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -71,7 +75,65 @@ public StateStore getStateStore(final String name) {
 
         final StateStore global = stateManager.getGlobalStore(name);
         if (global != null) {
-            return global;
+            return new KeyValueStore() {
+                private final KeyValueStore underlying = (KeyValueStore) global;
+
+                @Override public Object get(final Object key) {
+                    return underlying.get(key);
+                }
+
+                @Override public KeyValueIterator range(final Object from, final Object to) {
+                    return underlying.range(from, to);
+                }
+
+                @Override public KeyValueIterator all() {
+                    return underlying.all();
+                }
+
+                @Override public long approximateNumEntries() {
+                    return underlying.approximateNumEntries();
+                }
+
+                @Override public String name() {
+                    return underlying.name();
+                }
+
+                @Override public void init(final ProcessorContext context, final StateStore root) {
+                    underlying.init(context, root);
+                }
+
+                @Override public void flush() {
+                    underlying.flush();
```

Comment:
`flush` is also a write operation. Can we throw `UnsupportedOperationException` here ?

Reply:
Fixed.